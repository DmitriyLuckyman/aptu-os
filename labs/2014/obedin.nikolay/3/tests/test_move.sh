#!/usr/bin/env bash

source $(dirname $0)/common.sh

test() {
    run ./import    $FS_DIR $FILES_DIR/file1 /file1
    run ./import    $FS_DIR $FILES_DIR/file2 /file2
    run ./mkdir     $FS_DIR /test

    run ./move      $FS_DIR /file1 /test/file1
    run ./move      $FS_DIR /file2 /test/file2
    run ./move      $FS_DIR /test /test2
    run ./ls        $FS_DIR /
    run ./ls        $FS_DIR /test2

    run ./move      $FS_DIR /test2 /test2

    run ./mkdir     $FS_DIR /test2/test
    run ./move      $FS_DIR /test2 /test2/test/test

    run ./move      $FS_DIR /test2 /non_existing_dir/whatever
    run ./move      $FS_DIR /non_existing_dir/whatever /whatever
    run ./move      $FS_DIR /non_existing_dir /whatever
}
