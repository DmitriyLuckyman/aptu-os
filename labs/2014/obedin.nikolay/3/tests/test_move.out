D /
D test2

D test2
F file1
F file2

Error: can not copy/move into the same place
Error: can not move dir into itself
Error: destination file or dir not found
Error: source file or dir not found
Error: source file or dir not found
